### ISSUE TITLE
[ECR] [Feature request]: ECR registry access for kubernetes nodes that are outside the account 

### ISSUE THREAD

**Issue by Miller-5** [aws/containers-roadmap#1479]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
An option to connect to ECR registry with kubernetes cluster nodes that are outside of the aws account

**Which service(s) is this request for?**
ECR

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**

My task was to give EKS nodes a pull access to all private repositories in ECR that is on a different account
There is no out of the box solution to give full ECR access for eks nodes that are on a different aws account due to a missing feature
However you can give full access to a specific repository in ECR

I tried to add an assume-role for the node group role to get access on the other account, which should work on paper however I still couldn't pull any images from ECR
I tried a lot of other solutions and configurations to solve this but they didn't worked
The long version of the missing feature in kubernetes I got from our friend from aws support is:
"By default kubernetes will use the direct role of the instance to authenticate to ECR and not assume any roles even when these additional roles are available on the node IAM instance profile role"

basically the feature request is to have the option to give access to ECR for nodes that are outside of the ECR aws account 

**Are you currently working around this issue?**

The possible workarounds for now are:
1. Replicate the ECR repos to the EKS account
2. Add to each ECR repo a policy that will give access to the nodes (we can maybe create a bash script that will do it for us)
3. Manually use awscli in a node to login to the ecr repo, however the access will eventually expire and we would need to relogin every time (assume role, docker login and ecr login can all expire so its not recommended)

**Additional context**
Thank you!!

**Comment by toricls** [comment]
You can do this today, this [StackOverflow post](https://stackoverflow.com/a/57302321) would help you to have proper setup 🙌
